### PR TITLE
Use full clone for tests to enable proper new-code detection on SonarQube

### PR DESCRIPTION
**Pull request by arseniy-sonar**
~Excluding the file from coverage analysis for now to avoid SQ blocking PRs that do not touch the code at all. (for some reason, since this file was merged despite the failing QG on SQ, other PRs that target master are assumed to have changed this file and therefore also fail QG)~

### CODE REVIEW

**Review by frederic-tingaud-sonarsource — CHANGES_REQUESTED** [review]
I'm not sure about that. The other PR is failing because you merged the master in it and the analysis is on the most recent changelist and not the whole branch.
It should be possible to silence the error by squashing the PR.

**Comment by arseniy-sonar** [comment]
Indeed, probably there is an issue with SQ configuration or the warning it displays (Shallow clone detected during the analysis. Some files will miss SCM information. This will affect features like auto-assignment of issues. Please configure your build to disable shallow clone.)
Will have a look after I investigate the frontend_tests hanging.

File: .cirrus.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ env:
   SONAR_SCANNER_VERSION: 4.6.1.2450
   SONAR_SCANNER_HOME: ${HOME}/.sonar/sonar-scanner-${SONAR_SCANNER_VERSION}-linux
   PATH: ${SONAR_SCANNER_HOME}/bin:$PATH
-  CIRRUS_CLONE_DEPTH: 1
+  CIRRUS_CLONE_DEPTH: 0
```
Comment (marco-antognini-sonarsource) [inline .cirrus.yml:10]: 0 is the default value, so removing the line should work too. But if you prefer to keep it for explicitness that's fine I would say.

However, the duplicated lines L82 + L93 should probably be removed.

(I'm not sure it's worth the headache to set it to `1` for jobs that currently work fine...)

Comment (marco-antognini-sonarsource) [reply .cirrus.yml:10]: Nice fix BTW :-)

**Review by marco-antognini-sonarsource — APPROVED** [review]
LGTM

**Review by frederic-tingaud-sonarsource — APPROVED** [review]

**Pull request merged**